src/librustc/diagnostics.rs

```diff
@@ -791,8 +791,7 @@ trait Foo {
     fn bar(&self);
 }
 
-// we now declare a function which takes an object with Foo trait implemented
-// as parameter
+// we now declare a function which takes an object implementing the Foo trait
```

Comment:
I think this is correct, and it sounds better anyways. Having *which* and *that* in a sentence sounds awkward.